### PR TITLE
ci: auto label prs

### PR DESCRIPTION
**Pull request by joroshiba**
- Adds a workflow to apply PR labels to encourage small PRs
- Moves generated code into more clearly labelled directory
- Ensures generated code isn't included in line counts
- Auto labels prs based on config directories.

### CODE REVIEW

**Comment by SuperFluffy** [comment]
Bit torn on this because I feel like it's just going to generate extra noise?

**Review by SuperFluffy — APPROVED** [review]
Let's try this.

**Review by steezeburger — APPROVED** [review]

File: .github/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+proto:
```
Comment (SuperFluffy) [inline .github/labeler.yml:1]: Could we add a label for the containerfiles (== docker) and name it `container`? The docker-related workflows might also fall under it (less sure about that).

File: crates/astria-proto/src/lib.rs [hunk 2]
```diff
@@ -3,7 +3,7 @@
 #[rustfmt::skip]
 mod proto;
 
-pub use proto::tonic::{
+pub use proto::generated::{
```
Comment (SuperFluffy) [inline crates/astria-proto/src/lib.rs:6]: Nit: fine with the rename, although it feels orthogonal to the rest of the PR

File: .gitattributes [hunk 3]
```diff
@@ -0,0 +1 @@
+crates/astria-proto/src/proto/generated/** linguist-generated=true
```
Comment (SuperFluffy) [inline .gitattributes:1]: Never saw this, cool

**Review by SuperFluffy — APPROVED** [review]
Looks good. The auto labelling is a nice touch because I often forget stuff.

File: crates/astria-proto/src/lib.rs (hunk 2 above)
Comment (joroshiba) [reply crates/astria-proto/src/lib.rs:6]: It is kinda orthogonal, but related. 

I exclude generated lines from line changes but it wasn't clear that's what was happening. This increases that clarity.
